force-app/main/default/classes/OrderTriggerHandler.cls

```diff
@@ -0,0 +1,198 @@
+            }
+        }
+        
+        //get asset Record
+        List<Asset> assetList = [SELECT Id, Name, Status, Actual_Ship_Date__c FROM Asset WHERE Name LIKE :assetNameSet];
+        for(Asset asset : assetList){
+            asset.Status = 'Shipped';
+            asset.Actual_Ship_Date__c = Date.today();
+        }
+        update assetList;
+    }
+    
+    public static void addContactsToDripCampagin(List<Order> orderList){
+        set<Id> opportunityIdSet = new set<Id>();
+        for(Order order: orderList){
+            if(order.Status == 'Shipped' && order.OpportunityId != null){
+                opportunityIdSet.add(order.OpportunityId);
+            }
+        }
+        List<OpportunityContactRole> opportunityContactRoleList = [SELECT Id, Contact.Id, Contact.Name, Contact.Email, Contact.DripCampaign__c, OpportunityId, Role FROM OpportunityContactRole WHERE ROle='Primary Support'and OpportunityId IN :opportunityIdSet and Contact.DripCampaign__c = false];
+        
+        Set<Id> contactIdSet = new Set<Id>();
+        for(OpportunityContactRole opportunityContactRole : opportunityContactRoleList){
+            contactIdSet.add(opportunityContactRole.Contact.Id);
+        }
+        List<Contact> contactList = [SELECT Id, DripCampaign__c FROM Contact WHERE Id IN :contactIdSet];
+        for (Contact contact : contactList) {
+            contact.DripCampaign__c = true;
+        }
+        
+        update contactList;
+    }
+    
+    public static void checkUserRecords(List<Order> orderList) {
+        Set<Id> oppIdSet = new Set<Id>();
+        for (Order order : orderList) {
+            if (order.Status == 'Shipped' && order.OpportunityId != null) {
+                oppIdSet.add(order.OpportunityId);
+            }
+        }
+        
+        List<OpportunityContactRole> opportunityContactRoleList = [SELECT Id, Contact.Id, Contact.Name, Contact.Email, OpportunityId, IsPrimary,Role FROM OpportunityContactRole WHERE Role = 'Primary Support' and OpportunityId IN :oppIdSet and Contact.Email != null];
+        
+        Map<String, String> usernameToNameMap = new Map<String, String>();
+        for (OpportunityContactRole opportunityContactRole : opportunityContactRoleList) {
+            usernameToNameMap.put(opportunityContactRole.Contact.Email, opportunityContactRole.Contact.Name);
+        }
+        
+        if (usernameToNameMap.size() == 0) return;
+        
+        Profile customerCommunityLoginUserProfile = [SELECT Id, Name FROM Profile WHERE Name = 'Customer Community Login User' LIMIT 1];
+        
+        List<User> userList = [SELECT Id, Name, Email, Username, IsActive FROM User WHERE Username IN :usernameToNameMap.keySet()];
+        Set<Id> userIdSet = new Set<Id>();
+        List<User> newUserList = new List<User>();
+        for (String username : usernameToNameMap.keySet()) {
+            Boolean isUserExists = false;
+            for (User iteratedUser : userList) {
```

Comment:
try to avoid inner loop 
reffer this smaple code
`List<String> myStrings = new List<String>{'a', 'b'};
Boolean result = myStrings.contains('z');
System.assertEquals(false, result);`
Link: https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_methods_system_list.htm#apex_System_List_contains